### PR TITLE
Add minimal Docker setup to enable a build and test run

### PR DESCRIPTION
**Pull request by kjsanger**
This setup is sufficient to build and run tests inside a VScode container. There is one test which fails inside a Docker container because it invokes the docker client itself. Fixing that is tech debt

### CODE REVIEW

**Comment by mksanger** [comment]
The apt repository only contains 4.3.0 (latest) iRODS-dev / icommands / runtime packages

**Comment by kjsanger** [comment]
> The apt repository only contains 4.3.0 (latest) iRODS-dev / icommands / runtime packages

Not so, it has all the old ones, e.g. `https://packages.irods.org/apt/pool/bionic/main/i/`

You might have to set `IRODS_VERSION` to whatever package version they're using, if it's not just a basic x.y.z number.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  irods-server:
+    container_name: irods-server
+    image: "wsinpg/ub-18.04-irods-${IRODS_VERSION:-4.2.11}:${DOCKER_TAG:-latest}"
+    restart: always
+    ports:
+      - "1247:1247"
+      - "20000-20199:20000-20199"
+
+  baton:  
+    image: baton
```
Comment (mksanger) [inline docker-compose.yml:11]: The packages in the irods apt repo are built for amd architecture, so platform should be specified for macs with arm processors:
```suggestion
    image: baton
    platform: linux/amd64
```

Comment (kjsanger) [reply docker-compose.yml:11]: Thank you!

**Comment by mksanger** [comment]
Is the failing test expected to hang?  I get:
```
baton@294a95796d36:/workspace$ make check
Making check in src
make[1]: Entering directory '/workspace/src'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/workspace/src'
Making check in .
make[1]: Entering directory '/workspace'
make[1]: Leaving directory '/workspace'
Making check in tests
make[1]: Entering directory '/workspace/tests'
make  check_baton
make[2]: Entering directory '/workspace/tests'
make[2]: 'check_baton' is up to date.
make[2]: Leaving directory '/workspace/tests'
make  check-TESTS
make[2]: Entering directory '/workspace/tests'
make[3]: Entering directory '/workspace/tests'
```
Which then hangs, and the only log message is `Failed to open 'no_such_path': error 2 No such file or directory`, which I assume is expected.

**Comment by kjsanger** [comment]
The whole test suite hangs? If so, no. That doesn't happen on Linux or Intel macOS.

**Comment by mksanger** [comment]
The hang was due to not running `iinit` before testing.

**Pull request merged**